src/SW_Weather.c

```diff
@@ -527,6 +548,16 @@ void readAllWeather(
               weather_prefix
             );
         }
+
+        // Check to see if actual vapor pressure needs to be calculated
+        if(!SW_Weather.has_vp && SW_Weather.use_relHumidityMonthly) {

```

Comment:
- remove `SW_Weather.has_vp` from the condition `if(!SW_Weather.has_vp && SW_Weather.use_relHumidityMonthly) {` -- either we use monthly humidity inputs and then we need to calculate vapor pressure -- or, we use daily inputs for humidity related variables and then vapor pressure is calculated inside `_read_weather_hist()` in any case
- also we cannot calculate vapor pressure here if needed because of monthly humidity inputs because of the situation when the weather generator is in use and `allHist[yearIndex]->temp_avg[day]` is missing;
the best location that I can currently think of is inside `finalizeAllWeather()` just after the call to `generateMissingWeather()`

Reply:
Would we not need to add `SW_Weather.has_hurs` in place of `!SW_Weather.has_vp`? Since we calculate actual vapor pressure with `actualVaporPressure1()` then as well. But this will need to be backed up in some way making sure something of higher priority (e.g., min/max hurs, ...) is not given.

Reply:
`hurs` is a daily input and all cases of calculating `vp` from daily input needs to happen by `_read_weather_hist()` [see also https://github.com/DrylandEcology/SOILWAT2/pull/338#discussion_r1066200121]-- calculating `vp` is needed here (respectively later [see second bullet point of my comment]) if we use monthly humidity values